requests_kerberos/kerberos_.py

```diff
@@ -11,10 +11,29 @@
 from requests.structures import CaseInsensitiveDict
 from requests.cookies import cookiejar_from_dict
 
-from .exceptions import MutualAuthenticationError
+from .exceptions import MutualAuthenticationError, KerberosExchangeError
 
 log = logging.getLogger(__name__)
 
+# a log sniffing context manager for catching logged errors on a specific call
+class SniffLog(logging.Handler):
+    def __init__(self, log_to_sniff, sniff_min_level=logging.ERROR):
+        logging.Handler.__init__(self)
+        self.log_to_sniff = log_to_sniff
+        self.stored_logs = []
+
+        self.setLevel(sniff_min_level)
+
+    def __enter__(self, **kwargs):
+        self.log_to_sniff.addHandler(self)
+        return self
+
+    def __exit__(self, type, value, traceback):
+        self.log_to_sniff.removeHandler(self)
+
+    def emit(self, record):
+        self.stored_logs.append(record.getMessage())
+
```

Comment:
Yeah, it'll just require collecting those details directly (probably still using a context manager, just not one that's bound to the logger). I'll take a crack at that.